downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile.java

```diff
@@ -0,0 +1,413 @@
+        if (dates.get(goaLine) == null) {
+            dates.put(goaLine, instanceDate);
+        } else {
+            if (instanceDate > dates.get(goaLine)) {
+                dates.put(goaLine, instanceDate);
+            }
+        }
+        return instanceDate;
+    }
+
+    // Generates the third column of the GOA line by checking attributes of the referenceEntity instance.
+    private static String getSecondaryIdentifier(GKInstance referenceEntityInst) throws Exception {
+        if (referenceEntityInst.getAttributeValue(ReactomeJavaConstants.secondaryIdentifier) != null) {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.secondaryIdentifier).toString();
+        } else if (referenceEntityInst.getAttributeValue(ReactomeJavaConstants.geneName) != null) {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.geneName).toString();
+        } else {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.identifier).toString();
+        }
+    }
+
+    // Checks if this PhysicalEntity is one that is made of multiple PhysicalEntity instances.
+    private static boolean multiInstancePhysicalEntity(SchemaClass physicalEntitySchemaClass) {
+        if (physicalEntitySchemaClass.isa(ReactomeJavaConstants.Complex) || physicalEntitySchemaClass.isa(ReactomeJavaConstants.EntitySet) || physicalEntitySchemaClass.isa(ReactomeJavaConstants.Polymer)) {
+            return true;
+        }
+        return false;
+    }
+
+    // Returns all constituent PhysicalEntity instances of a Multi-instance PhysicalEntity
+    private static Set<GKInstance> getMultiInstanceSubInstances(GKInstance physicalEntityInst) throws Exception {
+        SchemaClass physicalEntitySchemaClass = physicalEntityInst.getSchemClass();
+        String subunitType = null;
+        Set<GKInstance> subInstanceProteins = new HashSet<>();
+        if (physicalEntitySchemaClass.isa(ReactomeJavaConstants.Complex)) {
+            subunitType = ReactomeJavaConstants.hasComponent;
+        } else if (physicalEntitySchemaClass.isa(ReactomeJavaConstants.Polymer)) {
+            subunitType = ReactomeJavaConstants.repeatedUnit;
+        } else if (physicalEntitySchemaClass.isa(ReactomeJavaConstants.EntitySet)) {
+            subunitType = ReactomeJavaConstants.hasMember;
+        }
+        for (GKInstance subunitInst : (Collection<GKInstance>) physicalEntityInst.getAttributeValuesList(subunitType)) {
+            subInstanceProteins.addAll(getBiologicalProcessProteins(subunitInst));
+        }
+        return subInstanceProteins;
+    }
+
+    // Checks that the incoming EntitySet instance only has EWAS members
+    private static boolean onlyEWASMembers(GKInstance entitySetInst) throws Exception {
+        Collection<GKInstance> memberInstances = entitySetInst.getAttributeValuesList(ReactomeJavaConstants.hasMember);
```

Comment:
Since this is looking to see if there are members and all members are EWASs, you could use test with the `allMatch` method.  If there are no members, it will return false:
```
return memberInstances
	.stream()
	.allMatch(member -> member.getSchemClass().isa(ReactomeJavaConstants.EntityWithAccessionedSequence));
```